### DISCUSSION TITLE
Self Host Khoj Cloud?

### DISCUSSION THREAD

**Discussion by KeithIMyers** [khoj-ai/khoj#526] [General]
Good Morning,
I know Klog cloud is a WIP but I wanted to know if there were any plans to allow it to be self hosted. My reasoning for asking is I often take a lower powered device with me when I am not at home and would love to have a way to securely access my content from any device. I have a server in a colo that I already use to run several docker containers and run my internal git server and the specs should be high enough to run a kloj instance.

**Comment by sabaimran** [comment]
Hi @KeithIMyers! Yes, for sure this is meant to be easily self-hostable. Our initial release is going to be using Google OAuth for login, for which you'd need a [Google developer account](https://developers.google.com/workspace/guides/create-credentials) and a test application.

I think we'll add a native sign-in mechanism as well so the Google dependency isn't there for accounts, but that might not be available in the initial release. That being said, would love to have your feedback on the dev experience when/if you do try hosting it on your own server!

**Comment by KeithIMyers** [comment]
Thank you @sabaimran - I would be definately be interested in that, I am also fine with the Google Sign-in as I have done a lot with that in the past. Do you have a idea on when a MVC will be for this and what the minimum requirements might be?

**Reply by debanjum** [reply]
Hey @KeithIMyers, we [released](https://github.com/khoj-ai/khoj/releases/tag/1.0.0) the upgraded Khoj today. You should be able to self-host it on your colo server by following the [self-hosting instructions](https://docs.khoj.dev/#/setup)

If you know how to setup Google Sign-in, you just need to set Google Client Id and Google Secret for your hosted Khoj to have Google OAuth based sign-in.

You can then use Khoj with the new Desktop, Obsidian, Emacs or Web client by switching their server URL to your Khoj server URL.

In terms of system requirements. If you choose to use Offline chat, it'd be ideal if you have GPU with 16GB VRAM for faster responses. But a machine with no GPU and 16GB RAM should suffice with online or offline chat. 
Note: _That RAM required to index your data shouldn't be too high now as the new Khoj use Postgres DB instead of loading your content index onto memory._ 

Let us know if you hit any blockers setting up a self-hosted Khoj or using it?

**Reply by KeithIMyers** [reply]
Awesome news, I will try this in a test VM later this week. How are the API keys configured for Google OAuth?

**Reply by debanjum** [reply]
You'll need to set the `GOOGLE_CLIENT_ID`, `GOOGLE_CLIENT_SECRET` environment variables before starting Khoj to use Google OAuth.
